### PR TITLE
Allow using https requests with proxy

### PR DESCRIPTION
**Pull request by alexeygrinko-xo**
https://xohellobar.atlassian.net/browse/CE-69

I didn't remove `appendBaseElement` method intentionally. Finally, it's an independent method which theoretically might be used in the future or outside of CrazyEgg repo.

### CODE REVIEW

**Comment by alexeygrinko-xo** [comment]
@imtall, @san650 or @mike-ce , pls merge it so I can continue using it in CE repo

**Comment by alexeygrinko-xo** [comment]
@imtall, I see all previous changes were accompanied with a version update in package JSON and creating new version tag. Shouldn't we do it here as well?

**Comment by imtall** [comment]
@alexeygrinko-xo  - I don't know. Probably :) @san650?

**Comment by san650** [comment]
@alexeygrinko-xo Yes, we are creating tags so it's easier to manage on `crazyegg` side.

See https://github.com/CrazyEggInc/crazyegg/wiki/Recording%3A-bundling-req.js

> After you finish your changes on ce-vdom project, increase the version of the package in the package.json file and create a new git tag. Use the 2.0.x format for the new version.

**Comment by san650** [comment]
@alexeygrinko-xo Looking at the PR I think we can remove the `appendBaseElement` method because after this change we're going to use the proxy for all URLs but I guess we'll still need the `findBaseNode` to calculate the base URL of the page.

Can you remove it?

**Comment by alexeygrinko-xo** [comment]
@san650, ok, will remove. Why do you think we need `findBaseNode`? For sites that include BASE tag on their own?

**Comment by alexeygrinko-xo** [comment]
@imtall, I don't have access to edit this repository, so I can't create a new release tag. That's also why I forked it before creating pull-request. Can you provide me with edit rights?

**Comment by imtall** [comment]
@alexeygrinko-xo  - you should have access now.

File: test/vdom-ext-test.js [hunk 1]
```diff
@@ -119,316 +119,270 @@ describe('vdom-ext', function() {
...
+    proxyUrl + "https:/test.com/img.jpg",
+    proxyUrl + "https:/test.com/img.jpg%3Fwidth=100&height=200",
     proxyUrl + "http:/test.com/img.jpg",
     proxyUrl + "http:/test.com/img.jpg%3Fwidth=100&height=200",
     proxyUrl + "http:/test.com/img.jpg",
     proxyUrl + "http:/test.com/img.png"
   ];
 
   describe('#vNodeCleanupUrls()', function() {
-    describe('when replace all is true', function() {
```
Comment (san650) [inline test/vdom-ext-test.js]: You're right, `replaceAll` is not needed.

**Comment by san650** [comment]
@alexeygrinko-xo I think `findBaseNode` will be useful to calculate the base URL of the page on the player. As you said, pages can have a <base> tag defined so we need to take that into account.

**Comment by alexeygrinko-xo** [comment]
@san650, replaced that method with `getBaseUrl`

**Comment by alexeygrinko-xo** [comment]
@imtall, I have access now, thank you.

**Pull request merged**